### PR TITLE
Refactor our test building infrastructure

### PR DESCRIPTION
**Pull request by stephenswat**
Currently, the way we build our tests is a bit of a confused mess, mainly because it has grown organically over time. This commit is an attempt to reorder the test building code in such a way that it becomes easier to read, and easier to work with.

I've renamed the test files to something more consistent, I've made sure that we compile all tests for a module into a single file, I've removed the main functions from individual tests, and I've borrowed a testing function from the vecmem build system.

Pinging our resident CMake guru @krasznaa on this one. :smile:

### CODE REVIEW

**Comment by stephenswat** [comment]
This MR should be ready to merge in, if someone wants to stamp their approval on it. :smile:

**Review by krasznaa — APPROVED** [review]
Fine, let's keep that `SOURCES` argument then... :thinking:

**Comment by stephenswat** [comment]
Wait hold on

**Comment by stephenswat** [comment]
That should be gone :thinking:

File: tests/CMakeLists.txt [hunk 1]
```diff
@@ -1,25 +1,27 @@
...
-    target_link_libraries(${TESTNAME} traccc::tests::common)
-    target_link_libraries(${TESTNAME} ${PLUGIN_LIBRARY})
-    set_target_properties(${TESTNAME} PROPERTIES FOLDER tests)
-    add_test(NAME unit_test_${TESTNAME} COMMAND ${TESTNAME})
-    set_tests_properties(unit_test_${TESTNAME} PROPERTIES
-    ENVIRONMENT TRACCC_TEST_DATA_DIR=${PROJECT_SOURCE_DIR}/data/)
-endmacro()
+    # Create the test executable.
+    set(TEST_NAME "test_${name}")
+    add_executable(${TEST_NAME} ${ARG_SOURCES})
```
Comment (krasznaa) [inline tests/CMakeLists.txt:7]: For me it's still very much here... :confused:

Comment (stephenswat) [reply tests/CMakeLists.txt:7]: I think I'm going mad.

**Comment by stephenswat** [comment]
Okay, let's pretend that never happened. _Now_ it's ready.

**Review by krasznaa — APPROVED** [review]
Yeah, let's push this in finally as well. :+1:

**Pull request merged**